lib/parser/prescan.cc

```diff
@@ -295,6 +295,11 @@ bool Prescanner::MustSkipToEndOfLine() const {
 void Prescanner::NextChar() {
   CHECK(*at_ != '\n');
   ++at_, ++column_;
+  while (at_[0] == '\xef' && at_[1] == '\xbb' && at_[2] == '\xbf') {
```

Comment:
It looks odd to me that this is a "while" rather than an "if".  Is it possible that there are multiple UTF-8 markers in a sequence in the file?  Also, are there similar byte sequences that would cause us to switch to encodings other than UTF-8?

Reply:
The UCS BOM is a "non-advancing space character" and we should skip over it wherever it appears in a UTF-8-encoded source file, so that the concatenation of two files with leading BOMs would still work.  Handling inline BOMs anywhere in the source would be hard and I haven't tried to deal with them yet.  But this loop allows one to prepend a redundant BOM to a file, at least.

I don't know of other markers that make manifest any other encoding that we'd support.